core/health/impl/src/main/java/org/opennms/core/health/impl/DefaultHealthCheckService.java

```diff
@@ -95,8 +93,8 @@ public DefaultHealthCheckService(BundleContext bundleContext) {
         try {
             // Fail if no checks are available
             List<HealthCheck> checks = getHealthChecks();
-            if (tag != null){
-                checks = checks.stream().filter(check -> check.getTags().contains(tag)).collect(Collectors.toList());
+            if (tags != null && tags.stream().anyMatch(tag -> !Strings.isNullOrEmpty(tag))){
```

Comment:
Add a test case on DefaultHealthCheckService to verify that it can process multiple tags properly